### DISCUSSION TITLE
Getting error `TypeError: 'type' object is not subscriptable` when trying to build on Linux

### DISCUSSION THREAD

**Discussion by im-coder-lg** [pyinstaller/pyinstaller#7869] [Help] [answered]
Hi,

My friends and I are trying to make an app with Python and Tkinter.

Repository for checking code: https://github.com/Futura-Py/weather

We have run into a hitch with one of our PRs. We are trying to add the [Sun-Valley-Ttk](https://github.com/rdbende/Sun-Valley-ttk-theme) theme to the app. We have done it before, with one of our previous projects, and it did generate an executable for all OSes. But now, we have added it to our PR, along with other features on our roadmap, and the Linux build is failing. Of course, we are using GitHub Actions to do this since it is more easier to manage and debug for us. 

Actions workflow log:

```sh
...
File "<frozen importlib._bootstrap_external>", line 522, in _check_name_wrapper
  File "<frozen importlib._bootstrap_external>", line 1027, in load_module
  File "<frozen importlib._bootstrap_external>", line [85](https://github.com/Futura-Py/weather/actions/runs/5878618475/job/15941054747?pr=16#step:4:86)2, in load_module
  File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
  File "<frozen importlib._bootstrap>", line 702, in _load
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 848, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/runner/work/weather/weather/extra-hooks/hook-sv_ttk.py", line 4, in <module>
    datas: list[tuple[str, str]] = collect_data_files("sv_ttk")
TypeError: 'type' object is not subscriptable
```

Workflow link: https://github.com/Futura-Py/weather/actions/runs/5878618475/job/15941054747?pr=16#step:4:7

Our PyInstaller [hook file](https://github.com/Futura-Py/weather/blob/ui-updates/extra-hooks/hook-sv_ttk.py)

```python
# extra-hooks/hook-sv_ttk.py
from PyInstaller.utils.hooks import collect_data_files

datas: list[tuple[str, str]] = collect_data_files("sv_ttk")
```

Our `.spec` file for Linux and macOS(we keep separate ones for Darwin and Linux, and separate for Windows since they seem to have some different configurations):

<details>

<summary>spec file</summary>

File at: https://github.com/Futura-Py/weather/blob/ui-updates/weather.spec

```spec
# -*- mode: python ; coding: utf-8 -*-
import os

block_cipher = None

a = Analysis(
    ["main.py"],
    pathex=[],
    binaries=[],
    datas=[],
    hiddenimports=[],
    hookspath=["extra-hooks"],
    hooksconfig={},
    runtime_hooks=[],
    excludes=[],
    win_no_prefer_redirects=False,
    win_private_assemblies=False,
    cipher=block_cipher,
    noarchive=False,
)

pyz = PYZ(a.pure, a.zipped_data, cipher=block_cipher)

exe = EXE(
    pyz,
    a.scripts,
    a.binaries,
    a.zipfiles,
    a.datas,
    [],
    exclude_binaries=False,
    name="Weather",
    debug=False,
    bootloader_ignore_signals=False,
    strip=False,
    upx=True,
    console=False,
    disable_windowed_traceback=False,
    argv_emulation=False,
    target_arch=None,
    codesign_identity=None,
    entitlements_file=None,
)

app = BUNDLE(
    exe,
    name="Weather.app",
    icon=None, # TODO: Make icon
    bundle_identifier=None,
)
```

</details>

We don't understand what went wrong. Windows and macOS have zero issues regarding the build, but Linux's Python throws an error saying that. Does anybody know the best fix for this?

**Comment by bwoodsend** [comment, marked as answer]
The `list[type]` syntax didn't exist until Python 3.9. Your Linux build is using Python 3.8.10. Either remove the type annotations or use a newer version of Python.

**Reply by im-coder-lg** [reply]
Oh, let me fix that quickly. Thanks!

**Reply by im-coder-lg** [reply]
Phew, it works! Thanks!!!
